src/credentials/FabricTable.h

```diff
@@ -351,14 +321,53 @@ class ConstFabricIterator
 class DLL_EXPORT FabricTable
 {
 public:
+    class DLL_EXPORT FabricTableDelegate
+    {
+    public:
+        FabricTableDelegate() {}
+        virtual ~FabricTableDelegate() {}
+
+        /**
+         * Gets called when a fabric is deleted, such as on FabricTable::Delete().
+         **/
+        virtual void OnFabricDeletedFromStorage(FabricTable & fabricTable, FabricIndex fabricIndex) = 0;
+
+        /**
+         * Gets called when a fabric is loaded into Fabric Table from storage, such as
+         * during FabricTable::Init().
+         **/
+        virtual void OnFabricRetrievedFromStorage(FabricTable & fabricTable, FabricIndex fabricIndex) = 0;
+
+        /**
+         * Gets called when a fabric in Fabric Table is persisted to storage, such as
+         * on FabricTable::AddNewFabric().
+         **/
+        virtual void OnFabricPersistedToStorage(FabricTable & fabricTable, FabricIndex fabricIndex) = 0;
+
+        /**
+         * @brief If set to true, the delegate must be deleted on removal from fabric table,
+         *        since the caller relinquished ownership of a dynamically allocated object.
+         *
+         * @param deleteOnRemoval - true for a delete on removal, false if owned by caller
+         */
+        void SetMustDeleteOnRemoval(bool deleteOnRemoval) { mDeleteOnRemoval = deleteOnRemoval; }
+
+        bool MustDeleteOnRemoval() const { return mDeleteOnRemoval; }
+
+        FabricTableDelegate * next = nullptr;
```

Comment:
Does this need to be public?
Should this be an intrusive list instead?

Reply:
It is an intrusive list, which is why it's public. Before it was private, but a friend of the other class making use of it --> actually public.

This code usage precedes using the more common "intrusive list".